pycode/memilio-simulation/memilio/simulation/bindings/models/osecir.cpp

```diff
@@ -48,98 +48,75 @@
 #include "pybind11/pybind11.h"
 #include "pybind11/stl_bind.h"
 #include "Eigen/Core"
+#include <algorithm>
+#include <cstddef>
 #include <vector>
 
 namespace py = pybind11;
 
 namespace
 {
 
-//select only the first node of the graph of each run, used for parameterstudy with single nodes
-template <class Sim>
-std::vector<Sim> filter_graph_results(
-    std::vector<mio::Graph<mio::SimulationNode<double, Sim>, mio::MobilityEdge<double>>>&& graph_results)
-{
-    std::vector<Sim> results;
-    results.reserve(graph_results.size());
-    for (auto i = size_t(0); i < graph_results.size(); ++i) {
-        results.emplace_back(std::move(graph_results[i].nodes()[0].property.get_simulation()));
-    }
-    return std::move(results);
-}
-
 /*
  * @brief bind ParameterStudy for any model
  */
-template <class Simulation>
+template <class Sim>
 void bind_ParameterStudy(py::module_& m, std::string const& name)
```

Comment:
I would move bind_ParameterStudy into a generic binding function (like bind_CoompartmentalModel), that returns the binding object. And then move the model specific definitions of run[...] functions into the model namespace. So you could call '.def(...)' on the return of bind_ParameterStudy.